### PR TITLE
ZON-2996: Prevent deleting published content

### PR DESCRIPTION
**Pull request by witsch**
See [ZON-2996](https://zeit-online.atlassian.net/browse/ZON-2996).

### CODE REVIEW

File: src/zeit/cms/repository/browser/menu.py [hunk 1]
```diff
@@ -29,8 +29,15 @@ class Delete(zeit.cms.browser.menu.LightboxActionMenuItem):
...
     def visible(self):
-        return IRepositoryContent.providedBy(self.context) and \
-            not IPublishInfo(self.context).published
+        if not IRepositoryContent.providedBy(self.context):
+            return False
+        elif IPublishInfo(self.context).published:
+            return False
+        elif IFolder.providedBy(self.context):
+            for item in self.context.values():
+                if IPublishInfo(item).published:
```
Comment (witsch) [inline src/zeit/cms/repository/browser/menu.py]: @wosc, is there a way to optimize this, i.e. ask webdav if there is any item with the 'published' property?

Comment (witsch) [inline src/zeit/cms/repository/browser/menu.py]: ftr, we've worked around this in c9cb0d600d1516580c3f87515791dd3eeff181f9 — now only the actual 'delete' view performs that check, but not the (much more frequently used) main view.

**Comment by witsch** [comment]
translation(s) have been added in https://github.com/ZeitOnline/zeit.locales/commit/72dd190bbd640f17f3d1e4e07b503cdfb5f584e0, so `zeit.locales` should be released as well…

File: src/zeit/cms/repository/browser/menu.py (hunk 1 above)
Comment (wosc) [inline src/zeit/cms/repository/browser/menu.py]: @witsch In theory: yes. The DAV server stores metadata as a triple `(path, namespace+name, value)`. So one should be able to do something like `SELECT path FROM facts WHERE url like "myfolder/*" AND name="published" AND value="yes"`.

However, as mentioned before, we can't/won't talk SQL directly, but rather go through yet another HTTP extension called the "query server", and I don't know offhand whether a) that allows the `url` in the where clause at all, and b) if it supports "startswith" or "like" or similar.

[Example queryserver usage](https://github.com/ZeitOnline/zeit.newsletter/blob/master/src/zeit/newsletter/category.py#L119), the API is unfortunately really quirky, for example you have to mention all variables you want to use in the where clause in the select clause, otherwise it silently returns no results.

Bottom line, since you've managed to cordon off that usecase (and it's a rare one), it might not be worth digging into this...
